chapter-09/recipe-04/cxx-example/CMakeLists.txt

```diff
@@ -9,12 +9,49 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+# we default to Release build type
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Release")
+endif()
+
 # for testing we will need the python interpreter
 find_package(PythonInterp REQUIRED)
 
-# we require boost::python and python development headers
+# we require python development headers
 find_package(PythonLibs ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR} EXACT REQUIRED)
-find_package(Boost REQUIRED COMPONENTS python)
+
+# now search for the boost component
+# depending on the boost version it is called either python, python27, python36, python37, ...
+
+if(PYTHON_VERSION_MAJOR VERSION_LESS 3)
```

Comment:
Early nitpick. Can this be done without the conditional?
```
list(
    APPEND _components
      python${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}
      python${PYTHON_VERSION_MAJOR}
      python
   )
```

Reply:
Good suggestion! Will fix this.